### ISSUE TITLE
Feature: Docs: JavaScript Styles

### ISSUE THREAD

**Issue by misterdai** [twbs/bootstrap#22764]
TLDR; List any CSS classes that each Bootstrap JavaScript code functionality adds, for easier optimisation.

It would be useful to have a list of styles applied by Bootstraps JavaScript code. Even after customising the CSS produced by Bootstrap, to only cover the functionality you may be using, there are further optimisations that could be made such as using [uncss](https://github.com/giakki/uncss).  The only issue with this is that styles that are only applied by JavaScript are not covered.  It's simple to add a list of styles to ignore, but without easily knowing which are made use of, it takes a lot of trial and error or tracking down.

**Comment by Johann-S** [comment]
I'm not sure about the use of `uncss` because Bootstrap possess a lots of CSS classes which aren't used in our JavaScript and it's pretty hard to list all the CSS classes available.

But I think it's a good idea to list CSS classes used by our Plugins. They are listed in our JS code in the object `ClassName`

**Comment by misterdai** [comment]
I use `uncss` as a final step for my static based personal website, which is using the v4 alpha of Bootstrap at the moment.  Works really well as it'll process all my static HTML files (can do live site requests as well) for styles used and remove any that I don't from the optimised CSS.  So it wouldn't need a complete list of CSS classes, only the ones added through the JS code.

Useful tip about the `ClassName` object holding the styles, I'll use that for now :)

**Comment by XhmikosR** [comment]
I was thinking that eventually we should prefix all of our classes with something like `bs-foo`. But that's for the future.

**Comment by XhmikosR** [comment]
On the other hand purgecss seems to work much better.

**Comment by mdo** [comment]
Closing as stale. PRs welcome here if someone wanted to take a stab at it.
